src/Mutex/Impl/Internal/MutexActivities.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Temporalio.Activities;
+using Temporalio.Client;
+using Temporalio.Workflows;
+
+namespace TemporalioSamples.Mutex.Impl.Internal;
+
+internal class MutexActivities
+{
+    private static readonly string RequestLockSignalName =
+        WorkflowSignalDefinition.FromMethod(
+                typeof(MutexWorkflow).GetMethod(nameof(MutexWorkflow.RequestLockAsync))
+                ?? throw new InvalidOperationException($"Method {nameof(MutexWorkflow.RequestLockAsync)} not found on type {typeof(MutexWorkflow)}"))
+            .Name ?? throw new InvalidOperationException("Signal name is null.");
+
+    private readonly ITemporalClient client;
+
+    public MutexActivities(ITemporalClient client)
+    {
+        this.client = client;
+    }
+
+    [Activity]
+    public async Task SignalWithStartMutexWorkflowAsync(SignalWithStartMutexWorkflowInput input)
+    {
+        var activityInfo = ActivityExecutionContext.Current.Info;
+
+        await this.client.StartWorkflowAsync(
+            (MutexWorkflow mw) => mw.RunAsync(null),
+            new WorkflowOptions(input.ResourceId, activityInfo.TaskQueue)
```

Comment:
I think the workflow ID should have something prepended just in case they are using this resource ID as a workflow ID elsewhere